src/order/basic.lean

```diff
@@ -11,13 +11,20 @@ import data.prod
 
 ## Definitions
 
+- `order_dual α` : A type tag reversing the meaning of all inequalities.
+
 ### Predicates on functions
 
-- `monotone f`: A function between two types equipped with `≤` is monotone if `a ≤ b` implies
+* `monotone f`: A function between two types equipped with `≤` is monotone if `a ≤ b` implies
   `f a ≤ f b`.
-- `strict_mono f` : A function between two types equipped with `<` is strictly monotone if
+* `antitone f`: A function between two types equipped with `≤` is antitone if `a ≤ b` implies
+  `f b ≤ f a`.
+* `strict_mono f` : A function between two types equipped with `<` is strictly monotone if
   `a < b` implies `f a < f b`.
-- `order_dual α` : A type tag reversing the meaning of all inequalities.
+* `strict_anti f` : A function between two types equipped with `<` is strictly antitone if
+  `a < b` implies `f b < f a`.
+* `strict_mono_on f s`: Same as `strict_mono f`, but for all `a, b ∈ s`.
```

Comment:
could you also add definitions `monotone_on` and `antitone_on` just for completeness?

Reply:
This one was quite a big change, because I had to restructure all to ~~please my perfectionism~~ efficiently replicate the API.